<!DOCTYPE html>
<html>
  <head>
    <title>Analysis of high-throughput sequencing data from NIH GEO database</title>
    <meta charset="utf-8">
    <meta name="author" content="Taavi Päll" />
    <link href="libs/remark-css/default.css" rel="stylesheet" />
    <link href="libs/remark-css/default-fonts.css" rel="stylesheet" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Analysis of high-throughput sequencing data from NIH GEO database
### Taavi Päll
### 26-05-2018

---




class: inverse, middle, center

# Introduction

---

## Signal vs noise

- Making discoveries is technically dependent on the ability of our methods to identify signal from overhelming amount of noise

- This pertains also in case of omics experiments (eg. gene expression analysis, proteomics) when thousands of measurements and hypotheses are made and tested in parallel

- Small proportion of found effects are real and scientifically interesting. How to sort out real effects from false positives?     

- To keep false positive discoveries under control different multiple comparison adjustments of P values are used (eg. Bonferroni), in case of omics experiments less conservative approaches are **FDR** and **qvalues**. 

- In order to use FDR and qvalues, p values should come from continuous distribution with "flat" shape

---

## Reproducibility

- Because generally only "successful" experiments get published, P values show bias towards zero in scientific articles and their distribution is truncated after 0.05. 
- So are truncated also reported omics experiments and only "top tables" with small number of significantly changed genes/features

- Peamiseks tulemuseks on üks-kaks geeni, mis on "edukalt" läbinud ka valideerimise ortogonaalsetes katsetes    

--

- Tulemused tunduvad huvitavad..., kas tasuks replitseerida&lt;sup&gt;1&lt;/sup&gt;? 
- Kuidas hinnata sellise eksperimendi kvaliteeti, kus on testitud paralleelselt tuhandeid hüpoteese?

.footnote[[1] Replitseerimine -- uute andmete korjamine kontrollimaks teaduslikke väiteid. Reprodutseerimine keskendub eelkõige andmete analüüsi korratavus. Reprodutseeritavus toetab, kuid ei taga teaduslike väidete paikapidavust.]

---

## QC checklist

1. Reproducibility: is data usable for reanalysis? (deposition to a public
database)

1. Compliance with GEO database submission requirements?

1. How many of the GEO submissions provide p value tables?

1. P histograms – how many experiments are poorly designed/analysed?

---

class: inverse, middle, center

# P values

---

## P values have uniform distribution

- P values were designed to follow uniform distribution in order control the number of false positive discoveries (type-I error rate)

--

- Under the null hypothesis, test statistic T has the distribution `\(F(t)\)` (eg. standard normal). p-value `\(P = F(T)\)` has a [probability distribution](https://stats.stackexchange.com/questions/10613/why-are-p-values-uniformly-distributed-under-the-null-hypothesis#10617): 
`$$Pr(P&lt;p) = Pr(F(T)&lt;p) = Pr(T&lt;F^{-1}(p)) = F(F^{-1}(p)) = p$$`

--

- The p-value is uniformly distributed:
      - when the null hypothesis is true
      - test statistic comes from continuous distribution

"The whole point of using the correct distribution (normal, t, f, chisq, etc.) is to transform from the test statistic to a uniform p-value. If the null hypothesis is false then the distribution of the p-value will (hopefully) be more weighted towards 0." ['Greg snow'](https://stats.stackexchange.com/a/10617)
---

## Test statistic from continuous distribution

Shape of the distribution is not influenced by your hypothesis but parameters ([Murdoch, 2008](https://www.tandfonline.com/doi/abs/10.1198/000313008X332421))

`$$H_0: \mu \le 0 \\ H_a: \mu \gt 0$$`

&lt;img src="img/murdoch2008-fig2.png" width="60%" style="display: block; margin: auto;" /&gt;

---

## Test statistic from discrete distribution

P value distribution is not uniform anymore and cannot be used for FDR adjustment..

&lt;img src="img/murdoch2008-fig3.png" width="60%" style="display: block; margin: auto;" /&gt;

---

## Distribution of P values 

&lt;img src="index_files/figure-html/unnamed-chunk-3-1.png" style="display: block; margin: auto;" /&gt;

---

background-image: url(http://varianceexplained.org/figs/2014-12-15-interpreting-pvalue-histogram/plot_melted-1.png)
class: center, bottom

.footnote[
Pilt: Typical P value histograms 
@[Variance explained](http://varianceexplained.org/statistics/interpreting-pvalue-histogram/).
]

???

Image credit: [Variance explained](http://varianceexplained.org/statistics/interpreting-pvalue-histogram/)

---

background-image: url(img/ncbi-geo.png)
background-size: cover
class: middle, center


# HT-seq datasets at Entrez GEO database

---

## GEO series submission dynamics and associated publications of HT-seq data

- We queried the NCBI GEO database for `"expression profiling by high throughput sequencing"`
- Query retrieved 13575 datasets
- First HT-seq data set was submitted to GEO database at 2006-09-08
- 68% of GEO data sets (9194) contain data from mouse or human origin 

&lt;img src="img/fig1.png" width="60%" style="display: block; margin: auto;" /&gt;

---

## GEO data submission rules

- We sought to determine the compliance with the formal requirements for GEO data submission (https://www.ncbi.nlm.nih.gov/geo/info/seq.html)    

- This pertains to the reproducibitly of data analysis    

- GEO archive has three required components   
    1. metadata, 
    1. processed data, and 
    1. raw data
        
---

## Publications and supplementary files

- 88% of human and mouse data sets are linked to their primary publication. 
- The proportion of datasets with publications is lower when other taxa are used as model organisms 78%
      
- Almost all GEO datasets (13560/13575) contain supplementary files

- The data sets lacking public supplementary files nevertheless show 73% publication rate which is similar to data sets with supplementary files


---

### Filename extensions and supplementary file names

- RAW.tar files are included in 57% (7722) of GEO series, whereas only RAW.tar and filelist.txt is present in 46% (6201) GEO series
- 'RAW.tar' archive contains one file per sample and samples can be directly mapped to metadata by using sample name
- Other common files contain annotations ‘readme.txt’ and processed data: commonly sequence read counts, gene expression analysis results or ChIP-Seq peak counts

&lt;img src="img/fig2.png" width="75%" style="display: block; margin: auto;" /&gt;

---

## Which supplementary files contain P values?

- Downloading all 34136 GEO supplementary files was not practical

- Sequence read counts can be either as raw read counts or normalised read counts
      - Gene expression analysis of HT-seq data is commonly performed by fitting models to discrete sequence read count data
      - Normalised sequence read counts are used for plotting and visual evaluation of the analysis results, but cannot be used to directly reproduce gene expression analysis

- To reduce number of files to be downloaded from GEO database, based on file extensions and filenames, we excluded from the dataset files that most plausibly don’t contain final processed data in a tabular format, such as binary files, annotations, raw data, graphics, and other non tabular files 

---

## Filter

- Filetype filter reduced the number of files by 60% and GEO series by 51%   
- Resulting dataset consisted of 13526 files from 6660 GEO series to be downloaded    

&lt;img src="img/fig5.png" style="display: block; margin: auto;" /&gt;

.footnote[Arrows from left to right, (1) GEO sets with supplementary files, (2) filter out RAW.tar and filelist.txt files from supplementary file names, (3) supplementary files with flat table formats that were downloaded, (4) tables with p values. Number of p value sets is the number of sets with &gt;4000 features.]

---

class: inverse, middle, center

# Results

---

## Number of samples and features of HT-seq experiments

- Mean number of samples was 27 and median 8. Most frequent sample size was 6     

- Mean number of features was 41935 and median 21899     

&lt;img src="img/fig6.png" style="display: block; margin: auto;" /&gt;

---

## Proportion of true null hypotheses in P value sets

&lt;img src="img/fig7.png" style="display: block; margin: auto;" /&gt;

- The number of [direct target genes of Myc](http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0001798) in human cells totals approximately 16% of all human genes
- [p53 has 346-399 validated target genes](https://www.nature.com/articles/onc2016502) (up to 3500, based on HT experiments)
- The number of high confidence `\(\beta\)`-[catenin target genes is 193](https://bmcgenomics.biomedcentral.com/articles/10.1186/1471-2164-15-74) (up to 1501 and 2389 genes)
---

## Top journals publishing HT-seq data

&lt;img src="img/fig8.png" style="display: block; margin: auto;" /&gt;

---

## Citations versus the number of true nulls

&lt;img src="img/fig9.png" style="display: block; margin: auto;" /&gt;

---

##  P value histogram types

&lt;img src="img/table1.png" style="display: block; margin: auto;" /&gt;

---

## Filtering non-informative features

&lt;img src="img/spark.png" style="display: block; margin: auto;" /&gt;

- [Tabel](spark.html)

---

## Conclusions

- Better reporting of HT-seq experiments analysis artefacts is needed and will improve reproducibility
- Based on available P value histograms, about 2/3 of experiments display deficencies in experiment design or statistical analysis
- Filtering out low expression level non-informative features is simplest way to fix or improve staistical inference in omics experiments 
- Reporting full sets of raw P value histograms should be compulsory when publishing omics experiment results

---

background-image: url(https://upload.wikimedia.org/wikipedia/commons/b/be/Sharingan_triple.svg)

## Tänud!

Ülo Maiväli ja Tanel Tenson


???

Image credit: [Wikimedia Commons](https://commons.wikimedia.org/wiki/File:Sharingan_triple.svg)
    </textarea>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function() {
  var d = document, s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})();</script>

<script>
(function() {
  var i, text, code, codes = document.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
})();
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
